### PR TITLE
Sphinx doc chapter 17

### PR DESCRIPTION
**Pull request by maximedenes**

### CODE REVIEW

**Review by silene — APPROVED** [review]

**Comment by Zimmi48** [comment]
A large paragraph was forgotten in the translation: it starts with `Now suppose we split the second pattern  of \texttt{length} into two`.

File: doc/sphinx/addendum/extended-pattern-matching.rst [hunk 1]
```diff
@@ -1,843 +1,611 @@
...
+length, by writing
+
+.. coqtop:: in
+
+   Fixpoint concat (n:nat) (l:listn n) (m:nat) (l':listn m) {struct l} :
+    listn (n + m) :=
+     match l in listn n, l' return listn (n + m) with
+     | niln, x => x
+     | consn n' a y, x => consn (n' + m) a (concat n' y m x)
+     end.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extended-pattern-matching.rst]: It seems that there was a copy-paste problem here but this is not was was in the LaTeX refman.

File: doc/sphinx/addendum/extended-pattern-matching.rst [hunk 2]
```diff
@@ -1,843 +1,611 @@
...
-have to use @ syntax.
-
-\Rem this feature is incoherent with the fact that parameters cannot be caught and
-consequently is somehow hidden. For example, there is no mention of it in error messages.
-
-\asection{Pattern-matching and coercions}
+on the object(s) matched. But it may depend and the typical case is
+when we write a proof by induction or a function that yields an object
+of dependent type. An example of proof using match in given in Section
+8.2.3.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extended-pattern-matching.rst]: Reference is hardcoded.

File: doc/sphinx/addendum/extended-pattern-matching.rst [hunk 3]
```diff
@@ -1,843 +1,611 @@
...
+         | or_introl h => or_introl (LE m' n') (LES n m h)
+         | or_intror h => or_intror (LE n' m') (LES m n h)
+         end
+     end.
+
+In the example of :g:`dec`, the first match is dependent while the second
+is not.
+
+The user can also use match in combination with the tactic :tacn:`refine` (see
+Section 8.2.3) to build incomplete proofs beginning with a match
```
Comment (Zimmi48) [inline doc/sphinx/addendum/extended-pattern-matching.rst]: Reference is hardcoded.

**Pull request merged**